### PR TITLE
feat: Lesson12 Resubmission

### PR DESCRIPTION
**Pull request by javyW**
Adding the comments for lesson12

### CODE REVIEW

File: lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/javyenware/CsvDataLoader.java [hunk 1]
```diff
@@ -85,12 +85,21 @@ private void populateGuestsWithCheckouts(String filePath, List<LibraryGuestModel
         // Map<String, List<CheckoutModel>> checkoutsByGuestEmail = new HashMap<>(); starts a new hashmap (a hashmap is data strcuture used to store data as a key value pair.)
         // The difference between a hashmap and a map is that a hashmap is less flexible and uses a hashtable for fast access)( a hashtable is a collection of key-value pairs, for each value you have a unique key for it)
```
Comment (anthonydmays) [inline lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/javyenware/CsvDataLoader.java:86]: `HashMap` is just an implementation of the `Map` interface. There are other implementations of the interface available in Java including the `LinkedHashMap` implementation.

https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html

File: lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/javyenware/CsvDataLoader.java [hunk 2]
```diff
@@ -85,12 +85,21 @@ private void populateGuestsWithCheckouts(String filePath, List<LibraryGuestModel
...
+        // The reason we are skipping line one because the first line provides the structure of the rest of the data set. It only provides and serves as a guide for how to understand and parse the data on the rest of the rows below.
     Map<String, List<CheckoutModel>> checkoutsByGuestEmail = new HashMap<>();
     try (CSVReader reader =
-        new CSVReader(new FileReader(new ClassPathResource(filePath).getFile()))) {
+      new CSVReader(new FileReader(new ClassPathResource(filePath).getFile()))) {
       reader.skip(1);
       String[] line;
+      // ((line = reader. reader.readNext()) != null) this is the start off the while loop. Its saying keep reading each line untill its not equal to null.
+      // ! means not and = means equal, and null meaning returning nothing 
+      //  while ((line = reader.readNext()) != null) loop through each line untill there are no more lines left.
```
Comment (anthonydmays) [inline lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/javyenware/CsvDataLoader.java:102]: Do you understand what the `guests.forEach` is doing below on line 111? Please describe.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/javyenware/CsvDataLoader.java (hunk 2 above)
Comment (javyW) [reply lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/javyenware/CsvDataLoader.java:106]: forEach() is a method that allows you to iterate over each element in a collection.

File: lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/models/LibraryGuestModel.java
[the remark this answers is not in the document]
Comment (javyW) [reply lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/models/LibraryGuestModel.java:1]: I reverted changes to the file. Removed public Object getEmail method and public Object setCheckdOutItems method.

**Pull request closed without merging**